README.md
## Cabbit Art Collection

![Static Preview](preview.png)
![Animated Preview](preview.gif)
*Lyuba character by Svetlana Kushnariova*

### Description:

A collection of game art assets by, or based on work by, [Svetlana Kushnariova (Cabbit)](https://opengameart.org/user/15048).

### Licensing:

Unless otherwise specified, all artwork is licensed under [OpenGameArt.org Attribution (OGA BY) version 3.0](LICENSE.txt) or later. OGA BY explicitly allows re-licensing under [Creative Commons Attribution (CC BY)](docs/other-licenses/CC-BY-3.0.txt).

### Links:

- [OpenGameArt.org submission](https://opengameart.org/node/79804)
- Cabbit's submissions:
  - [24x32 characters with faces (big pack)](https://opengameart.org/node/24823)
  - [24x32 heroine Lyuba (sprites, faces, pictures)](https://opengameart.org/node/50909)
  - [24x32 characters, 16x16 tiles](https://opengameart.org/node/72969)
- diamonddmgirl's submissions:
  - [Edited and Extended 24x32 Character Pack](https://opengameart.org/node/66147)
  - [24x32 Black Character Pack](https://opengameart.org/node/72198)
  - [24x32 Characters Lying Down](https://opengameart.org/node/72611)
  - [24x32 Pepper&Carrot Characters](https://opengameart.org/node/74802)
- Sprite bases (not included):
  - [Original bases by Cabbit](https://opengameart.org/node/24944)
  - [Reworks by diamondmgirl](https://opengameart.org/node/67861)
  - [Cabbit Bases Collection by AntumDeluge](https://opengameart.org/node/79237)

### Notes:

Assets with filenames ending with "v" plus a number indicate an older version of the image/work.


CHANGES.txt
0.5:
- Separated outfit sprites into sub-directories.
- Renamed "healer" sprites to "cleric"
- Extracted more outfit parts.
- Created more recolors.
- Created townfolk-child-m-003 from townfolk-child-m-002.
- Removed CC BY licensed Pepper&Carrot sprites & facesets.

0.4
- Added more outfit recolors.
- Created outfits:
  - suit-002 from townfolk-adult-m-009
  - tuxedo-001 from suit-002 (custom bowtie & jacket)
- Created outfit parts:
  - bowtie-001
  - dress-006 from dress-001
  - dress-007 from dress-005
  - gi-001
  - necktie-001
  - outfit-005 from suit-002
- Extracted outfits:
  - outfit-006 from townfolk-adult-m-001
  - outfit-007 from townfolk-adult-m-009
- Extracted outfit parts:
  - dress-child-001 from townfolk-child-f-003 (diamonddmgirl)
  - hair-short-004 from townfolk-adult-m-003
  - hair-short-005 from townfolk-adult-m-002
  - hair-short-006 from townfolk-adult-m-004
  - hair-short-007 from fighter-m-001
  - hair-short-008 from fighter-m-002
  - jacket-001 from outfit-006
  - pants-001 from outfit-006
  - pants-002 from suit-002
  - shirt-001 from suit-002
  - shirt-002 from suit-002
  - shoes-001 from suit-002
  - shoes-002 from pants-001
  - shoes-child-001 from townfolk-child-f-003 (diamonddmgirl)
  - t-shirt-001 from outfit-006
  - vest-001 from suit-002
  - vest-002 from suit-002

0.3
- Applied scale2x filter on more sprites.
- Created sprites:
  - Santa-001
  - Santa-002
  - townfolk-bearded_man-001
  - townfolk-bearded_man-002
  - townfolk-mayor-m-001
- Extracted outfit parts from aristocrate-m-001:
  - beard-002
  - hat-005
  - santa_suit-001
  - santa_suit-002
  - santa_hat-001
  - suit-001
- Created outfit parts:
  - hair-balding-004 from hair-balding-001.
  - beard-003 from beard-001
  - beard-004 from beard-003
  - beard-005 from beard-004
  - beard-006 from beard-004
  - moustache-001 from beard-004
- Added black recolor of hair-balding-001.

0.2
- Extracted more facesets.
- Touched up some sprites & added GIMP source files.

0.1
- Initial collection for OpenGameArt.org.
- Added resources by Svetlana Kushnariova (Cabbit) & diamonddmgirl.


release.sh
#!/bin/sh

# Script for packaging release.
#
# This script is licensed under Creative Commons Zero (CC0).

VERSION=$1

README="Name:
  Cabbit Collection
Version:
  ${VERSION}
Description:
  A collection of sprites created by, or based on the work of, Svetlana Kushnariova (Cabbit).
Licensing:
  - OpenGameArt.org Attribution (OGA BY) version 3.0 or later (see: LICENSE-OGA-BY-3.0.txt)
  - Creative Commons Attribution (CC BY) version 3.0 or later (see: LICENSE-CC-BY-3.0.txt)
Copyright/Attribution:
  Created by Svetlana Kushnariova (Cabbit) <lana-chan@yandex.ru>, diamonddmgirl, & Jordan Irwin (AntumDeluge)
Links:
  - OpenGameArt.org page: https://opengameart.org/node/79804
  - For links to Cabbit's & diamonddmgirl's submissions, see "sources.md" file.
Notes:
  - IMPORTANT: As part of Cabbit's attribution requirements, please credit her as \"Svetlana Kushnariova\"
    & include her email address: lana-chan@yandex.ru"

SOURCES="### Cabbit Collection Sources:

<table style=\"border: 0px;\">
  <tr style=\"border: 0px;\">
    <td style=\"border: 0px; vertical-align: top; text-align: center;\">
      ![Static Preview](preview.png)
    </td>
    </tr>
    <tr style=\"border: 0px;\">
    <td style=\"border: 0px; vertical-align: top; text-align: center;\">
      ![Animated Preview](preview.gif)
    </td>
  </tr>
</table>

- (OGA BY 3.0 / CC BY 3.0)
  - [24x32 Black Character Pack](https://opengameart.org/node/72198) by Cabbit & diamonddgirl
  - [24x32 characters, 16x16 tiles](https://opengameart.org/node/72969) by Cabbit
  - [24x32 Characters Lying Down](https://opengameart.org/node/72611) by Cabbit & diamonddgirl
  - [24x32 characters with faces (big pack)](https://opengameart.org/node/24823) by Cabbit
  - [24x32 heroine Lyuba (sprites, faces, pictures)](https://opengameart.org/node/50909) by Cabbit
  - [Edited and Extended 24x32 Character Pack](https://opengameart.org/node/66147) by Cabbit & diamonddgirl"

if [ ! -z "${VERSION}" ]; then
	echo "Version: ${VERSION}"

	ZIP="cabbit-${VERSION}.zip"

	if [ -f "${ZIP}" ]; then
		echo "ERROR: \"${ZIP}\" release already exists, cannot continue"
		exit 1
	fi

	if [ ! -d "TEMP" ]; then
		mkdir "TEMP"
	fi

	echo -e "${README}" > "TEMP/README.txt"
	echo -e "${SOURCES}" > "TEMP/sources.md"
	cp "LICENSE.txt" "TEMP/LICENSE-OGA-BY-3.0.txt"
	cp "docs/other-licenses/CC-BY-3.0.txt" "TEMP/LICENSE-CC-BY-3.0.txt"

	cd "art"
	zip -r "../${ZIP}" ./ -x "/originals*" "*.DS_Store" "*.db"
	cd "../TEMP"
	zip -r "../${ZIP}" ./
	cd "../"
	zip -r "${ZIP}" "CHANGES.txt" "TODO.txt" "preview.png" "preview.gif" "preview/"

	echo
	echo "Cleaning up ..."
	find "TEMP" -type f -delete
	find "TEMP" -type d -empty -delete
else
	echo "ERROR: Must specify version as first argument"
	exit 1
fi
